system/include/emscripten/wire.h

```diff
@@ -53,22 +53,37 @@ namespace emscripten {
             typedef typename std::remove_cv<typename std::remove_reference<T>::type>::type type;
         };
 
+#if __has_feature(cxx_rtti)
+#define HANDLE_TYPEID(typeid_of, canonicalized_type) return &typeid(typeid_of);
+#else
+#define HANDLE_TYPEID(typeid_of, canonicalized_type) \
+    static_assert(!has_unbound_type_names, \
+        "Unbound type names are illegal with RTTI disabled. " \
+        "Either add -DEMSCRIPTEN_HAS_UNBOUND_TYPE_NAMES=0 to or remove -fno-rtti " \
+        "from the compiler arguments"); \
+    static_assert(!std::is_polymorphic<canonicalized_type>::value, \
+        "Canonicalized<T>::type being polymorphic is illegal with RTTI disabled");
+#endif
+
         template<typename T>
         struct LightTypeID {
             static constexpr TYPEID get() {
                 typedef typename Canonicalized<T>::type C;
-                return (has_unbound_type_names || std::is_polymorphic<C>::value)
-                    ? &typeid(C)
-                    : CanonicalizedID<C>::get();
+                if(has_unbound_type_names || std::is_polymorphic<C>::value) {
+                    HANDLE_TYPEID(T, C);
```

Comment:
`HANDLE_TYPEID` expands to a `static_assert`, which means that the `if(has_unbound_type_names || std::is_polymorphic<C>::value)` branch has no effect since the assert is evaluated at compile time. Perhaps a regular `assert()` was intended here?

Also, while on this, the repository convention is to have a space in `if (`.